### DISCUSSION TITLE
Can tinytroupe use openai o1-preview or o1-mini api?

### DISCUSSION THREAD

**Discussion by PoYuYang0416** [microsoft/TinyTroupe#46] [General]
I had tried tinytroupe, it is a fascinating  work. However, I need to use o1-preview or o1-mini for deeper insight for scientific works. I tried it but there is error.

**Comment by makedjakila** [comment]
Hi,

Tokens are very expensive with gpt-4o...priceless with o1-preview. Personally, I use gpt-4o-mini and it works fine.

Maybe you should double check the model ID you are using.
